mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -336,14 +361,19 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
     }
 
     val w = if (!isDefined(weightCol) || $(weightCol).isEmpty) lit(1.0) else col($(weightCol))
-    val instances: RDD[Instance] =
-      dataset.select(col($(labelCol)), w, col($(featuresCol))).rdd.map {
-        case Row(label: Double, weight: Double, features: Vector) =>
-          Instance(label, weight, features)
-      }
+    val offset = if (!isDefined(offsetCol) || $(offsetCol).isEmpty) {
+      lit(0.0)
+    } else {
+      col($(offsetCol)).cast(DoubleType)
+    }
```

Comment:
minor style comment - I think these 4 lines here need to be indented with 2 more spaces:
  lit(0.0)  
} else {  
  col($(offsetCol)).cast(DoubleType)  
}